comp/core/telemetry/telemetry.go

```
@@ -0,0 +1,200 @@
+	meterProvider *sdk.MeterProvider
+}
+
+func newRegistry() *prometheus.Registry {
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+	reg.MustRegister(collectors.NewGoCollector())
+	return reg
+}
+
+func newProvider(reg *prometheus.Registry) *sdk.MeterProvider {
+	exporter, err := promOtel.New(promOtel.WithRegisterer(reg))
+
+	if err != nil {
+		panic(err)
+	}
+
+	return sdk.NewMeterProvider(sdk.WithReader(exporter))
+}
+
+func newTelemetry() Component {
+	return &telemetryImpl{
+		registry:      registry,
+		meterProvider: provider,
+	}
+}
+
+// Same as `newTelemetry“ without the global.
+// Can be merged with `newTelemetry` when the global is removed
+func newMock() Component {
+	reg := prometheus.NewRegistry()
+	provider := newProvider(reg)
+
+	return &telemetryImpl{
+		registry:      reg,
+		meterProvider: provider,
+	}
+}
+
+// TODO (components): Remove this when all telemetry is migrated to the component
+func GetCompatComponent() Component {
+	return newTelemetry()
+}
+
+func (t *telemetryImpl) Handler() http.Handler {
+	return promhttp.HandlerFor(t.registry, promhttp.HandlerOpts{})
+}
+
+func (t *telemetryImpl) Reset() {
+	registry = prometheus.NewRegistry()
+	t.registry = registry
+}
+
+func (t *telemetryImpl) Meter(name string, opts ...metric.MeterOption) metric.Meter {
+	return t.meterProvider.Meter(name, opts...)
+}
+
+func (t *telemetryImpl) NewCounter(subsystem, name string, tags []string, help string) Counter {
+	return t.NewCounterWithOpts(subsystem, name, tags, help, DefaultOptions)
+}
+
+func (t *telemetryImpl) NewCounterWithOpts(subsystem, name string, tags []string, help string, opts Options) Counter {
+	name = opts.NameWithSeparator(subsystem, name)
+
+	c := &promCounter{
+		pc: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Subsystem: subsystem,
+				Name:      name,
+				Help:      help,
+			},
+			tags,
+		),
+	}
+	_ = t.registry.Register(c.pc)
```

Do you know what kind of error `Register` can  return?

---

[It can return several kinds of errors ](https://github.com/prometheus/client_golang/blob/main/prometheus/registry.go#L294). 

This used to be `MustRegister` which would panic on error. I can change it back if you prefer.

---

It is usually better to not ignore the errors. If `MustRegister` was used so I think it makes sense to keep using it.

---

I made the change in the follow up PR :)